app/javascript/components/program/index.js

```diff
@@ -1,18 +1,102 @@
 import React, { PureComponent } from "react";
 import Layout from 'antd/es/layout';
+const Sider = Layout.Sider;
 import LayoutHeader from "../common/LayoutHeader";
 import LayoutFooter from "../common/LayoutFooter";
+import MenuBar from "./MenuBar";
+import Section from "./Section";
 
 class Program extends PureComponent {
+
+    constructor() {
+      super()
+      this.state = {
+          title: "",
+          sections: [],
+          items: []
+      }
+    }
+
+    componentDidMount() {
+        var items = []
+
+        if (this.props.match.params.section_id) {
+          fetch("/manifesto_sections/" + this.props.match.params.section_id + ".json")
+          .then(res => res.json())
+          .then((data) => {
+              items = data.items
+          })
+          .catch(console.log)
+        }
+
+        fetch("/parties/" + this.props.match.params.party_acronym + "/manifesto.json")
+        .then(res => res.json())
+        .then(data => {
+          this.setState({
```

Comment:
you don't need to use the current state here. it is enough to do
```
this.setState({
    title: data.title,
    sections: data.sections,
    items: items
});
```

Reply:
done